### PR TITLE
cpu/cc2538: refine periph/spi implementation

### PR DESCRIPTION
**Pull request by smlng**
### Contribution description

Adapt the periph SPI implementation to make use of vendor header defines and also utilise proper
GPIO API functions instead of manually configure through registers.

### CODE REVIEW

File: boards/remote-pa/include/periph_conf.h [hunk 1]
```diff
@@ -66,17 +66,17 @@ static const spi_clk_conf_t spi_clk_config[] = {
...
     },
     {
-        .dev      = SSI1,
-        .mosi_pin = GPIO_PC7,
-        .miso_pin = GPIO_PA4,
-        .sck_pin  = GPIO_PB5,
+        .num      = 1,
+        .mosi_pin = GPIO_PIN(2, 7),
+        .miso_pin = GPIO_PIN(0, 4),
+        .sck_pin  = GPIO_PIN(2 ,5),
```
Comment (PeterKietzmann) [inline boards/remote-pa/include/periph_conf.h]: `GPIO_PB5` -> `GPIO_PIN(1, 5)`

File: cpu/cc2538/include/periph_cpu.h [hunk 2]
```diff
@@ -200,7 +205,7 @@ typedef struct {
  * @{
  */
 typedef struct {
-    cc2538_ssi_t *dev;      /**< SSI device */
+    uint8_t num;            /**< SSI device */
```
Comment (PeterKietzmann) [inline cpu/cc2538/include/periph_cpu.h]: The semantics of this parameter changed, but the documentation didn't. Can you extend the comment? For example, mention that this is the device **number** 1...x or so

File: cpu/cc2538/periph/spi.c [hunk 3]
```diff
@@ -69,33 +73,16 @@ void spi_init(spi_t bus)
...
-
-    IOC_PXX_OVER[spi_config[bus].mosi_pin] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].miso_pin] = IOC_OVERRIDE_DIS;
-    IOC_PXX_OVER[spi_config[bus].sck_pin ] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].cs_pin  ] = IOC_OVERRIDE_OE;
-
-    gpio_hardware_control(spi_config[bus].mosi_pin);
-    gpio_hardware_control(spi_config[bus].miso_pin);
-    gpio_hardware_control(spi_config[bus].sck_pin);
-    gpio_hardware_control(spi_config[bus].cs_pin);
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/spi.c]: `gpio_hardware_control` isn't used anywhere else. How about we remove its definition? Or do you plan to cleanup cc2538/gpio anyway?

File: cpu/cc2538/periph/spi.c [hunk 4]
```diff
@@ -26,40 +26,44 @@
...
+
 /**
  * @brief   Array holding one pre-initialized mutex for each SPI device
  */
 static mutex_t locks[SPI_NUMOF];
 
 static inline cc2538_ssi_t *dev(spi_t bus)
 {
-    return spi_config[bus].dev;
+    return (cc2538_ssi_t *)(SSI0_BASE + (spi_config[bus].num * 0x1000));
```
Comment (PeterKietzmann) [inline cpu/cc2538/periph/spi.c]: This function is called in every `acquire`, `transfer`, `release` function. Do you have any estimation how this change effects performance?

File: boards/remote-pa/include/periph_conf.h (hunk 1 above)
Comment (smlng) [reply boards/remote-pa/include/periph_conf.h]: oops, will fix

File: cpu/cc2538/periph/spi.c (hunk 4 above)
Comment (smlng) [reply cpu/cc2538/periph/spi.c]: then it will be constant lookup just as before.

**Comment by smlng** [comment]
Hi @PeterKietzmann thx for the quick feedback and testing:

* I like the `.num` device index better than actually providing the hardware address in `periph_conf.h` because the developer typically cares about which device is enabled and not its address. The `.num` is required if you want to use SPI1 only, that is not using/configuring SPI0. Contrary to the timer implementation this is way simpler here because there are no IRQ handlers that need to be matched to the SPI device used.
* yes the CS stuff I saw, too - but to me that's definitely a separate PR, also to keep this one simple.
* clock settings: I'll look into, this should be fixed here.

**Comment by smlng** [comment]
@PeterKietzmann are we good here? As said clock issue and CS hwpins will be extra PRs

**Comment by PeterKietzmann** [comment]
The new openmote needs adaption as well

**Comment by smlng** [comment]
done

**Comment by PeterKietzmann** [comment]
please squash

**Pull request merged**